ddtrace/commands/ddtrace_run.py

```diff
@@ -106,12 +99,9 @@ def main():
     log.debug("program executable: %s", executable)
 
     try:
-        # Raises OSError for permissions errors in Python 2
-        #        PermissionError for Python 3
         os.execl(executable, executable, *args.command[1:])
-    except (OSError, PermissionError):
-        print("ddtrace-run: executable '%s' does not have executable permissions.\n" % executable)
-        parser.print_usage()
-        sys.exit(1)
+    except Exception:
+        print("ddtrace-run: error launching '%s'" % executable)
```

Comment:
```suggestion
        print("ddtrace-run: error launching '%s'" % executable)
        parser.print_usage()
```

and we might want to add some examples to the usage string

Reply:
Also from a usability standpoint would it be better to have the print at the bottom of the output so it's the first thing the user sees?

Reply:
I don't think printing usage when the usage is correct, but something else is wrong, a good pattern.

Reply:
To put that into perspective, here's my thinking after 20 years on Unix:
```
$ cat --wqdwq
cat: illegal option -- -
usage: cat [-benstuv] [file ...]
$ touch x
$ chmod 000 x
$ cat x
cat: x: Permission denied
```

Reply:
Again, I'm thinking about common misusage of `ddtrace-run file.py` where the correct command was `ddtrace-run python file.py`. With this change, the user will get 
```
$ ddtrace-run file.py
ddtrace-run: error launching 'file.py'
<exception>
```
My concern is that this is a pretty simple error to make and it's not very clear was the remediation is from the program output

> I don't think printing usage when the usage is correct, but something else is wrong, a good pattern.

Agreed

Reply:
It's not a misusage:`ddtrace-run file.py` works if `file.py` is executable and has a shebang.
If it does not, then the printed exception will give the caller a hint: permission issue, shebang missing, etc.

Reply:
I get that and agree that the behaviour in that case makes sense but that's a different use-case than the one I'm concerned about. It might not be easy for us to tell the difference between these cases in order to print the usage in the case I'm thinking of and not in the case you mentioned.

I'd still be tempted to print the usage in this case since I think it's much more likely that when `ddtrace-run file.py` fails that the intention was to run `ddtrace-run python file.py` and not that the shebang was forgotten (presumably the user would have run `python file.py` before prefixing with `ddtrace-run`. I might be alone in this opinion though 😛

Reply:
Ok, I've added a case (Py3 only for now, I'm lazy) to handle that. I'm not 100% fan but at least it does a better job at hinting the user I hope.

This is still not 100% accurate, e.g. if the file has permissions 111 the suggestion will fail anyway. 🤷